test/integration_test.go

```diff
@@ -479,6 +479,34 @@ func TestTerragruntWorksWithExistingGCSBucket(t *testing.T) {
 	validateGCSBucketExistsAndIsLabeled(t, location, gcsBucketName, nil)
 }
 
+func TestTerragruntCorrectlyMirrorsTerraformGCPAuth(t *testing.T) {
+	t.Parallel()
+
+	// We need to ensure Terragrunt works correctly when GOOGLE_CREDENTIALS are specified.
+	// There is no true way to properly unset creds from the environment, but we still try
+	// to unset the CI credentials during this test.
+	defaultCreds := os.Getenv("GCLOUD_SERVICE_KEY")
+	defer os.Setenv("GCLOUD_SERVICE_KEY", defaultCreds)
+	os.Unsetenv("GCLOUD_SERVICE_KEY")
+	os.Setenv("GOOGLE_CREDENTIALS", defaultCreds)
```

Comment:
You might want to disable `t.Parallel` for this test to avoid colliding with other tests that might expect these env vars.

Reply:
That's a good idea.